services/util/util.go

```diff
@@ -53,12 +54,14 @@ type CommandRun struct {
 	ExitCode int
 }
 
-// There's only one now but compose into a builder pattern for Options so
-// it's easy to add more (such as dropping permissions, etc).
+// A builder pattern for Options so it's easy to add various ones (such as dropping permissions, etc).
 type cmdOptions struct {
 	failOnStderr bool
 	stdoutMax    uint
 	stderrMax    uint
+	env          []string
+	uid          *uint32 // Pointers so we can distinguish if it's been set.
```

Comment:
since you never let the caller construct these directly, you could probably just set these to defaults before you apply options:

```go
 o := cmdOptions{
   uid: os.Geteuid(),
   gid: os.Getgid()
}
for _, opt := range options {
   opt(o)
}

// use options here, always constructing a credential
```

Reply:
True. Still have to validate they differ because it's EPERM to call setuid/gid() even if they are the values you already are...

Reply:
Fixed